docs/user_guide/concepts/main_concepts/data_management.rst

```diff
@@ -9,70 +9,65 @@ Understand How Flyte Handles Data
 Types of Data
 =============
 
-There are two parts to the data in Flyte:
+In Flyte, data is categorized into metadata and raw data to optimize data handling and improve performance and security.
 
-1. Metadata
+* **Metadata**: Small values, like integers and strings, are treated as "stack parameters" (passed by value). This metadata is globally accessible to Flyte components (FlytePropeller, FlyteAdmin, and other running pods/jobs). Each entry is limited to 10MB and is passed directly between tasks. On top of that, metadata allow in-memory computations for branches, partial outputs, and composition of multiple outputs as input for other tasks.
 
-* It consists of data about inputs to a task, and other artifacts.
-* It is configured globally for FlytePropeller, FlyteAdmin etc., and the running pods/jobs need access to this bucket to get the data.
+* **Raw data**: Larger data, such as files and dataframes, are treated as "heap parameters" (passed by reference). Flyte stores raw data in an object store (e.g., S3), uploading it on first use and passing only a reference thereafter. Tasks can then access this data via Flyte’s automated download or streaming, enabling efficient access to large datasets without needing to transfer full copies.
 
-2. Raw data
+*Source code reference for auto-offloading value sizes limitation:*
 
-* It is the actual data (such as the Pandas DataFrame, Spark DataFrame, etc.).
-* Raw data paths are unique for every execution, and the prefixes can be modified per execution.
-* None of the Flyte control plane components would access the raw data. This provides great separation of data between the control plane and the data plane.
+.. raw:: html
 
-.. note:
-  Metadata and raw data can be present in entirely separate buckets.
+   <a href="https://github.com/flyteorg/flyte/blob/6c4f8dbfc6d23a0cd7bf81480856e9ae1dfa1b27/flytepropeller/pkg/controller/config/config.go#L184-L192">View source code on GitHub</a>
```

Comment:
@Future-Outlier Since `.rst` does not support remote link references, I am using HTML to force the use of the gitsha link.